zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -1,19 +1,33 @@
 //! Functions for initializing the various databases.
 
-use rusqlite::{types::ToSql, NO_PARAMS};
+use rusqlite::{params, types::ToSql, NO_PARAMS};
 
 use zcash_primitives::{
     block::BlockHash,
     consensus::{self, BlockHeight},
-    zip32::ExtendedFullViewingKey,
 };
 
-use zcash_client_backend::encoding::encode_extended_full_viewing_key;
+use zcash_client_backend::{
+    encoding::encode_extended_full_viewing_key, keys::UnifiedFullViewingKey,
+};
 
 use crate::{address_from_extfvk, error::SqliteClientError, WalletDb};
 
+#[cfg(feature = "transparent-inputs")]
+use {
+    zcash_client_backend::encoding::AddressCodec,
+    zcash_primitives::legacy::keys::IncomingViewingKey,
+};
+
 /// Sets up the internal structure of the data database.
 ///
+/// The database structure is the same irrespective of whether or
+/// not transparent spends are enabled, and it is safe to use a
+/// wallet database created without the ability to create transparent
+/// spends with a build that enables transparent spends (though not
+/// the reverse, as wallet balance calculations would ignore the
+/// prior transparent inputs controlled by the wallet).
```

Comment:
Clarify that this compatibility is between builds with and without the "transparent-inputs" feature; not between wallets created by this version of `zcash_client_sqlite` and wallets created by previous versions.